app/helpers/application_helper/button/button_new_discover_cloud.rb

```diff
@@ -0,0 +1,11 @@
+class ApplicationHelper::Button::ButtonNewDiscoverCloud < ApplicationHelper::Button::Basic
+  include ApplicationHelper::Button::Mixins::SubListViewScreenMixin
+
+  def visible?
+    !sub_list_view_screen?
+  end
+
+  def disabled?
+    ManageIQ::Providers::CloudManager.subclasses.select(&:supports_discovery?).count.zero?
```

Comment:
Does this need to be an Async call? @himdel @Fryguy

Reply:
If we can *guarantee* that `supports_discovery?` never talks to the provider, I don't think it needs to be async..

Reply:
By provider do you mean the codebase or the actual provider like their api? The `supports_discovery` refers to this https://github.com/ManageIQ/manageiq-providers-azure/pull/193/files#diff-56132469fdcb7ca60ae7bf6c2e7f107fR25 in `ManageIQ::Providers::Azure::CloudManager` and Amazon, etc.

Reply:
The former :) 

Agreed, looks like this is essentially a constant depending just on the class.. I think we can merge as is :)

Reply:
Cool, feel free to press the merge button @himdel. Thanks for your help.